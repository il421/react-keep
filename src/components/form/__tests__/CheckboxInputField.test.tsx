import { mount, ReactWrapper } from "enzyme";
import React from "react";
import { Form } from "react-final-form";

import {
  CheckboxInputField,
  CheckboxInputFieldProps
} from "../CheckboxInputField";

let props: CheckboxInputFieldProps,
  wrapper: ReactWrapper<CheckboxInputFieldProps, {}>;

beforeEach(() => {
  props = {
    className: "className",
    name: "name",
    id: "id"
  };
});

test("should render CheckboxInputField, and contain input element", () => {
  wrapper = mount<CheckboxInputFieldProps>(
    <Form initialValues={{}} onSubmit={() => {}}>
      {() => <CheckboxInputField {...props} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
  expect(wrapper.find("#id").exists()).toBeTruthy();
  expect(wrapper.find("label[htmlFor='id']").exists()).toBeTruthy();
});
