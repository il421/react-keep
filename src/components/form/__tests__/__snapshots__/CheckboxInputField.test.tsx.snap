// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CheckboxInputField, and contain input element 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <CheckboxInputField className=\\"className\\" name=\\"name\\" id=\\"id\\">
    <Field className=\\"className\\" name=\\"name\\" id=\\"id\\" type=\\"checkbox\\" component={[Function: FieldAdapter]}>
      <FieldAdapter input={{...}} meta={{...}} className=\\"className\\" id=\\"id\\" render={[undefined]}>
        <div className=\\"className\\">
          <input id=\\"id\\" name=\\"name\\" value={[undefined]} checked={false} onBlur={[Function]} onChange={[Function]} onFocus={[Function]} type=\\"checkbox\\" />
          <label htmlFor=\\"id\\" />
        </div>
      </FieldAdapter>
    </Field>
  </CheckboxInputField>
</ReactFinalForm>"
`;
